packages/host/app/services/local-realm.ts

```diff
@@ -48,19 +62,67 @@ export default class LocalRealm extends Service {
         }
         break;
       case 'available':
-        if (data.type === 'visitRequest') {
-          let { id, path, staticResponses } = data;
+        if (data.type === 'setEntryAcknowledged') {
+          if (!this.#setEntryDeferred) {
+            throw new Error(
+              `received setEntry acknowledgement without corresponding setEntry request`
+            );
+          }
+          this.#setEntryDeferred.fulfill();
+          return;
+        }
+        if (data.type === 'startFromScratch') {
+          if (!this.#fromScratch) {
+            throw new Error(
+              `the fromScratch runner has not been registered with the local realm service`
+            );
+          }
+          let { realmURL } = data;
+          let worker = this.state.worker;
+          this.#fromScratch(new URL(realmURL)).then((state) => {
+            send(worker, {
+              type: 'fromScratchCompleted',
+              state: serializeRunState(state),
+            });
+          });
+          return;
+        }
+        if (data.type === 'startIncremental') {
+          if (!this.#incremental) {
+            throw new Error(
+              `the incremental runner has not been registered with the local realm service`
+            );
+          }
+          let { prev, url, operation } = data;
           let worker = this.state.worker;
-          // might want to keep track of these promises for orderly tear down...
-          this.workerRenderer.visit(path, staticResponses, (html) =>
-            send(worker, { type: 'visitResponse', id, path, html })
+          this.#incremental(
+            deserializeRunState(prev),
+            new URL(url),
+            operation
+          ).then((state) =>
+            send(worker, {
+              type: 'incrementalCompleted',
+              state: serializeRunState(state),
```

Comment:
in this case incremental indexing may have _removed_ state.instances from the indexing (which setEntries never addressed), so we'll need a fresh copy of the instances sent back to the server